api/src/opentrons/drivers/rpi_drivers/gpio_simulator.py

```diff
@@ -27,48 +27,50 @@ def board_rev(self) -> BoardRevision:
         return self._board_rev
 
     @board_rev.setter
-    def board_rev(self, boardrev: BoardRevision):
+    def board_rev(self, boardrev: BoardRevision) -> None:
         self._board_rev = boardrev
 
     def _initialize(self) -> Dict[str, int]:
-        lines = {}
-        for pin in gpio_group.pins:
-            lines[pin.name] = pin.by_board_rev(self.board_rev)
+        name_pin_num = (
+            (pin.name, pin.by_board_rev(self.board_rev)) for pin in gpio_group.pins
+        )
+        # Dict comprehension eliminating None values.
+        lines = {k: v for (k, v) in name_pin_num if v is not None}
```

Comment:
Can we change this back to a for-in loop? Personally I find previous version more readable

```suggestion
        lines = {}
        for pin in gpio_group.pins:
            line = pin.by_board_rev(self.board_rev)
            if line:
                lines[pin.name]
```

Reply:
Ok